lib/src/kbasesearchengine/main/ObjectDecorator.java

```diff
@@ -0,0 +1,48 @@
+package kbasesearchengine.main;
```

Comment:
The purpose of this class is, it can be used to subclass if there are more than one decorator. Since we have only one decorator (ObjectDecorator) it may seem like a NOOP.

Reply:
But why is it necessary to have a class to subclass? If another decorator is needed just implement the interface directly.

Reply:
Removed the class